odd_collector_profiler/profilers/azure_sql/profiler.py

```diff
@@ -0,0 +1,72 @@
+from typing import Any, Dict, List, Tuple
+
+from funcy import lmap
+from odd_models.models import DataSetFieldStat, DataSetStatistics, DatasetStatisticsList
+from oddrn_generator import AzureSQLGenerator
+from sqlalchemy import text
+from sqlalchemy.engine import Connection
+
+from odd_collector_profiler.domain.config import AzureSQLConfig
+from odd_collector_profiler.domain.profiler import Profiler
+from odd_collector_profiler.domain.statistics.column_statistic import ColumnStatistic
+from odd_collector_profiler.helpers.sql_dialect import SQLDialect
+from odd_collector_profiler.profilers import DATA_PROFILER
+from odd_collector_profiler.services.data_frame_reader import TableDataframeReader
+from odd_collector_profiler.services.data_profiler import DataProfiler
+
+
+class AzureSQLProfiler(Profiler, SQLDialect):
+    """Profiler for Postgresql datasource"""
+
+    config_model = AzureSQLConfig
+    data_frame_reader = TableDataframeReader
+
+    def __init__(self, config: Dict[str, Any], data_profiler: DataProfiler):
+        super().__init__(config, data_profiler)
+        self.generator = AzureSQLGenerator(
+            host_settings=f"{self.config.server}.database.windows.net:{self.config.port}",
+            databases=self.config.database,
+        )
+
+    def get_statistics(self) -> DatasetStatisticsList:
+        items: List[DataSetStatistics] = []
+
+        with self.connect() as connection:
+            query = "SELECT name FROM sys.schemas WHERE principal_id=1"
```

Comment:
Explicit what is principal_id=1

Reply:
added comment:
# The principal_id field is an ID of the principal that owns this schema.